src/features/node-fetch.js

```diff
@@ -57,6 +57,9 @@ global.System.constructor.prototype.fetch = async url => {
         return { status: 500, statusText: e.toString() };
     }
   } else {
-    return fetch(url);
+    return fetch(url).then(function (res) {
+      // Workaround for https://github.com/npm/make-fetch-happen/issues/28
+      return res.status === 304 ? fetch(url, { cache: 'reload' }) : res;
```

Comment:
So, this effectively turns every "revalidate" into a double latency request since it will first send the revaldiation request, then immediately send a new request to get a 200. As a result it's double latency by default, which isn't ideal.

In addition, the only remaining cache case with this change is the case of there being no revalidation request - but even then I assume the headers won't be available either? That is we might still have an error case there.

Overall it seems the best approach for now might be just to revert back to node-fetch until this header caching is solved.